src/hevm/src/EVM/SMT.hs

```diff
@@ -465,7 +465,7 @@ exprToSMT = \case
   Min a b ->
     let aenc = exprToSMT a
         benc = exprToSMT b in
-    "(ite (<= " <> aenc `sp` benc <> ") " <> aenc `sp` benc <> ")"
+    "(ite (bvule " <> aenc `sp` benc <> ") " <> aenc `sp` benc <> ")"
```

Comment:
encoding here was using integer Leq instead of the bitvector variant, produced a type error when passed to the smt solver